core/build_playerglobal/src/lib.rs

```diff
@@ -62,3 +70,188 @@ pub fn build_playerglobal(
+            method
+        } else {
+            return;
+        };
+
+        if !is_native {
+            return;
+        }
+
+        // Note - technically, this could conflict with
+        // a method with a name starting with `get_` or `set_`.
+        // However, all Flash methods are named with lowerCamelCase,
+        // so we'll never actually need to implement a native method that
+        // would cause such a conflict.
+        let method_prefix = match trait_.kind {
+            TraitKind::Getter { .. } => "get_",
+            TraitKind::Setter { .. } => "set_",
+            _ => "",
+        };
+
+        let flash_to_rust_path = |path: &str| path.to_lowercase().replace('.', "::");
+
+        let mut path = "crate::avm2::globals::".to_string();
+
+        let trait_name = &abc.constant_pool.multinames[trait_.name.0 as usize - 1];
+
+        if let Some(parent) = parent {
+            // This is a method defined inside the class. Append the class namespace
+            // (the package) and the class name.
+            // For example, a namespace of "flash.system" and a name of "Security"
+            // turns into the path "flash::system::security"
+            let multiname = &abc.constant_pool.multinames[parent.0 as usize - 1];
+            path += &flash_to_rust_path(resolve_multiname_ns(&abc, multiname));
+            path += "::";
+            path += &flash_to_rust_path(resolve_multiname_name(&abc, multiname));
+            path += "::";
+        } else {
+            // This is a freestanding function. Append its namespace, which corresponds
+            // (the package).
+            // For example, the freestanding function "flash.utils.getDefinitionByName"
+            // has a namespace of "flash.utils", which turns into the path
+            // "flash::utils"
+            path += &flash_to_rust_path(resolve_multiname_ns(&abc, trait_name));
+            path += "::";
+        }
+
+        // Append the trait name - this corresponds to the actual method
+        // name (e.g. `getDefinitionByName`)
+        path += method_prefix;
+        path += resolve_multiname_name(&abc, trait_name);
+        println!("Rust path: {:?}", path);
+
+        // Now that we've built up the path, convert it into a `TokenStream`.
+        // This gives us something like
+        // `crate::avm2::globals::flash::system::Security::allowDomain`
+        //
+        // The resulting `TokenStream` is suitable for usage with `quote!` to
+        // generate a reference to the function pointer that should exist
+        // at that path in Rust code.
+        rust_paths.insert(method_id.0, TokenStream::from_str(&path).unwrap());
+    };
+
+    // We support three kinds of native methods:
+    // instance methods, class methods, and freestanding functions.
+    // We're going to insert them into an `FnvHashMap`, so the order
+    // that we visit them in doesn't matter.
+    for (i, instance) in abc.instances.iter().enumerate() {
```

Comment:
I'm not entirely sure how it's matters, but it seems like [`nativegen.py`](https://github.com/adobe/avmplus/blob/master/utils/nativegen.py#L1913) recursively traverses through `Script` -> `Method`/`Class`, rather than a linear `Instance` scan.

Reply:
Nothing depends on the iteration order, since we just use the method ids.